forum/qa-plugin/q2a-googleauthenticator-login/q2a-googleauthenticator-overrides.php

```diff
@@ -0,0 +1,68 @@
+<?php
+
+function qa_set_logged_in_user($userId, $handle = '', $remember = false, $source = null)
+{
+    require_once QA_INCLUDE_DIR . 'app/cookies.php';
+    qa_start_session();
+    if (!isset($userId)) {
+        // logout
+        qa_report_event(
+            'u_logout',
+            qa_get_logged_in_userid(),
+            qa_get_logged_in_handle(),
+            qa_cookie_get()
+        );
+        qa_clear_session_cookie();
+        qa_clear_session_user();
+
+        return;
+    }
+
+    // login
+    require_once QA_INCLUDE_DIR . 'db/selects.php';
+    $result = qa_db_read_all_assoc(
+        qa_db_query_sub(
+            'SELECT 2fa_enabled FROM ^users WHERE userid = #', $userId
+        )
+    );
+    if (count($result) !== 1) {
+        echo 'Invalid num_rows';
```

Comment:
nie wiem co to konkretnie ma robić, ale wyświetlenie gdziekolwiek takiego błędu nie brzmi dobrze - nic nie mówi użytkownikowi

Reply:
to powinna być zmodyfikowana kopia metody q2a. Widzę, że @Mariusz08 trochę mocno tu sobie wszedł z butami :) do zmiany raczej.